Classes/git/PBGitRevList.mm

```diff
@@ -129,12 +129,6 @@ - (void) addGitObject:(GTObject *)obj toCommitSet:(NSMutableSet *)set
 
 	NSAssert(commit, @"Can't add nil commit to set");
 
-	for (GTCommit *item in set) {
-		if ([item.OID isEqual:commit.OID]) {
-			return;
-		}
-	}
-
 	[set addObject:commit];
```

Comment:
Yep, sorry for not pointing that out explicitly. There's one last tidbit, but it requires changes to Objective-Git itself : `-[GTOID hash]` allocates an `NSData` object to perform the hashing, which (given that we have some sets and dictionaries with large sizes) has a tendency to show up in profiles.